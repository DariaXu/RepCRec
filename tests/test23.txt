// Test that younger transactions abort rather than wait for older ones.
begin(T1)
begin(T2)
begin(T3)
// Oldest to youngest.
begin(T4)     
W(T2, x1, 15)
// T1 older waits for T2 younger.
R(T1, x1)     
R(T2, x6)
// T3 younger dies since T2 is older.
W(T3, x6, 22) 
W(T4, x8, 12)
// T2 older waits for T4 younger.
R(T2, x8)     
// T4 younger dies since T2 is older.
R(T4, x1)     
end(T2)
end(T1)
dump()
// ---
// assertCommitted(T1)
// assertCommitted(T2)
// assertAborted(T3)
// assertAborted(T4)
