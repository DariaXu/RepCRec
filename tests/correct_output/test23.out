Site 2: T2 write x1=15
Transaction T1 blocked by a lock conflict. Locks: [T2]
T2 reads on Site 1 - x6: 60
Transaction T3 blocked by a lock conflict. Locks: [T2]
Site 1: T4 write x8=12
Site 2: T4 write x8=12
Site 3: T4 write x8=12
Site 4: T4 write x8=12
Site 5: T4 write x8=12
Site 6: T4 write x8=12
Site 7: T4 write x8=12
Site 8: T4 write x8=12
Site 9: T4 write x8=12
Site 10: T4 write x8=12
Transaction T2 blocked by a lock conflict. Locks: [T4]
Transaction T4 blocked by a lock conflict. Locks: [T2]
Abort: T4
T2 reads on Site 1 - x8: 80
Commit: T2
T1 reads on Site 2 - x1: 15
Commit: T1
Site 1: T3 write x6=22
Site 2: T3 write x6=22
Site 3: T3 write x6=22
Site 4: T3 write x6=22
Site 5: T3 write x6=22
Site 6: T3 write x6=22
Site 7: T3 write x6=22
Site 8: T3 write x6=22
Site 9: T3 write x6=22
Site 10: T3 write x6=22
Site 1 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 2 - x1: 15, x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x11: 110, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 3 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 4 - x2: 20, x3: 30, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x13: 130, x14: 140, x16: 160, x18: 180, x20: 200
Site 5 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 6 - x2: 20, x4: 40, x5: 50, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x15: 150, x16: 160, x18: 180, x20: 200
Site 7 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 8 - x2: 20, x4: 40, x6: 60, x7: 70, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x17: 170, x18: 180, x20: 200
Site 9 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 10 - x2: 20, x4: 40, x6: 60, x8: 80, x9: 90, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x19: 190, x20: 200
